fe/src/stores/useProjectStore.ts

```diff
@@ -245,51 +245,78 @@ export const useProjectStore = defineStore('project', {
       // add the selected coaches to data object
       this.selectedCoaches.forEach((coach: User) => coachList.push(coach.url))
 
-      const data = {
+      return {
         name: this.projectName,
         partnerName: this.projectPartnerName,
         extraInfo: this.projectLink,
         requiredSkills: skillsList,
         coaches: coachList,
       }
-
-      // POST request to make a project
+    },
+    submitProject(
+      skills: Array<ProjectTableSkill>,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      callback: any
+    ) {
+      const projectData = this.formatProjectData(skills)
       instance
-          .post('projects/', convertObjectKeysToSnakeCase(data))
-          .then((response) => {
-            this.loadProjects()
-
-            // clear fields when project is made successfully
-            this.isLoadingProjects=  false
-            this.projectName= ''
-            this.projectPartnerName= ''
-            this.projectLink= ''
-            this.filterCoaches=''
-            this.selectedCoaches= []
-            useSkillStore().loadSkills()
+        .post('projects/', convertObjectKeysToSnakeCase(projectData))
+        .then(() => {
+          this.loadProjects()
+          callback(true)
+        })
+        .catch(function (error) {
+          console.log(error)
+          callback(false)
+        })
+    },
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async getAndSetProject(id: string, callback: any) {
+      // console.log(id)
+      return instance.get('projects/' + id).then((data) => {
+        console.log(data)
```

Comment:
You have the right to remain silent, everything you `console.log()` can and will be used against you. 🚨